discord/channel.py

```diff
@@ -130,9 +130,9 @@ class TextChannel(discord.abc.Messageable, discord.abc.GuildChannel, Hashable):
         The category channel ID this channel belongs to, if applicable.
     topic: Optional[:class:`str`]
         The channel's topic. ``None`` if it doesn't exist.
-    position: :class:`int`
+    position: Optional[:class:`int`]
         The position in the channel list. This is a number that starts at 0. e.g. the
-        top channel is position 0.
+        top channel is position 0. Can be ``None`` if the channel was recieved in an interaction.
```

Comment:
I don't know how true this comment is, it can be removed if needed

Reply:
Official docs say that resolved interaction data returns partial channel objects and: 
> Partial Channel objects only have id, name, type and permissions fields.

So this is [expected to be] true.